clients/src/test/java/org/apache/kafka/clients/MetadataTest.java

```diff
@@ -378,26 +382,6 @@ public void testRejectOldMetadata() {
         }
     }
 
-    @Test
-    public void testMaybeRequestUpdate() {
```

Comment:
After removing the `updateLastSeenEpoch` private method, there wasn't much point in this test. The epoch update is covered better in other tests anyways

Reply:
Why would removal of a private method make this test less relevant? Which test case already covers this?

Reply:
Removing the private `updateLastSeenEpoch` isn't really relevant actually, you're right. This test was simulating the epoch getting updated by something other than the metadata response. This is now covered by the new test cases in FetcherTest, but that's an indirect test. 

I'll revert this and update the test

Reply:
I ended up writing a new test `testUpdateLastEpoch`